lib/cli/run.ts

```diff
@@ -8,80 +8,84 @@ import { generateReport, cleanupReport } from '../report';
 import { getConfig } from './config';
 import { Logger } from '../logger';
 import { waitFor } from '../utils/wait-for';
+import { adbInstall, adbLaunch } from '../utils/adb';
+import {
+  xcrunInstall,
+  xcrunLaunch,
+  xcrunRestore,
+  xcrunStatusBar,
+  xcrunUi,
+} from '../utils/xcrun';
```

Comment:
All of the separate build and run steps that use either ADB or xcrun have been refactored out into separate util functions. This is so that the same functions can be easily reused at other times, such as using `xcrunLaunch` or `adbLaunch` during testing as part of the `reload` method.